core/src/main/java/org/elasticsearch/cluster/routing/allocation/NodeRebalanceResult.java

```diff
@@ -73,16 +75,28 @@ public boolean isDeltaAboveThreshold() {
     }
 
     /**
-     * Returns the current weight of the node if the shard is not added to the node.
+     * Returns if the simulated weight of the shard after assigning the node to it would be better than the current node's weight.
+     * If {@code true}, then the shard should be rebalanced to the node, assuming allocation is allowed to the node.
      */
-    public float getCurrentWeight() {
-        return currentWeight;
+    public boolean isBetterWeightWithShardAdded() {
+        return betterWeightWithShardAdded;
     }
 
-    /**
-     * Returns the weight of the node if the shard is added to the node.
-     */
-    public float getWeightWithShardAdded() {
-        return weightWithShardAdded;
+    @Override
+    protected void innerToXContent(XContentBuilder builder, ToXContent.Params params) throws IOException {
+        builder.field("better_weight_than_current", betterWeightThanCurrent);
```

Comment:
I'm a bit confused here why we need 3 booleans (and also confused by the naming):

For rebalancing to occur, the target node (with shard added) must 1) have a better weight and 2) the difference in weight change must be above a certain threshold.

With this in mind, isn't it sufficient to just store 1 and 2 and derive rebalancability from these 2 facts?